### PR TITLE
RazorPageGenerator: extracted creating RazorEngine into a method [refactoring]

### PR DESCRIPTION
**Pull request by evil-shrike**
RazorPageGenerator: extracted creating RazorEngine into a method to simplify reusing logic in custom generators
RazorPageGenerator: added optional cli argument with base directory  (previously current dir always was used)

### CODE REVIEW

**Comment by rynowak** [comment]
@pranavkm can you take a look?

@evil-shrike - thanks for the PR. This is a tool that we use to produce some generated code that we ship as part of the diagnostics page (among others). There's no real support commitment from us, but we're happy to accept a changes like this if it makes your life better 😄

File: src/RazorPageGenerator/Program.cs [hunk 1]
```diff
@@ -52,8 +60,15 @@ public static IList<RazorPageGeneratorResult> MainCore(string rootNamespace, str
...
                 builder.Features.Add(new SuppressChecksumOptionsFeature());
+                if (configure != null)
+                {
+                    configure(builder);
+                }
             });
-
+            return razorEngine;
+        }
+        public static IList<RazorPageGeneratorResult> MainCore(RazorEngine razorEngine, string targetProjectDirectory)
```
Comment (pranavkm) [inline src/RazorPageGenerator/Program.cs]: New line before this.

File: src/RazorPageGenerator/Program.cs [hunk 2]
```diff
@@ -37,7 +45,7 @@ public static int Main(string[] args)
             return 0;
         }
 
-        public static IList<RazorPageGeneratorResult> MainCore(string rootNamespace, string targetProjectDirectory)
+        public static RazorEngine CreateRazor(string rootNamespace, Action<IRazorEngineBuilder> configure = null)
```
Comment (pranavkm) [inline src/RazorPageGenerator/Program.cs]: `CreateRazorEngine`

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request merged**